pre-commit

```diff
@@ -13,6 +13,9 @@ else
     AGAINST=4b825dc642cb6eb9a060e54bf8d69288fbee4904
 fi
 
+# *Only* test the prospective commit: ignore unstaged changes.
+git stash save -q --keep-index
```

Comment:
I'm just wondering about problems caused by doing this on a branch with no changes:
 8.3.x  alex: ~/dev/drupal > git stash save -q --keep-index
 8.3.x  alex: ~/dev/drupal > git stash list
 8.3.x  alex: ~/dev/drupal > git stash pop -q
No stash found.

Imo we should only pop if something is stashed.